Parser/pegen.c

```diff
@@ -2678,26 +2684,24 @@ deal_with_gstring2(Parser *p, Token* a, asdl_expr_seq* raw_expressions, Token*b)
             if (item == NULL) {
                 return NULL;
             }
-        } 
+        }
         asdl_seq_SET(seq, i, item);
     }
 
-    char* _str = PyBytes_AsString(b->bytes);
-    if (_str == NULL) {
-        return NULL;
-    }
-    PyObject* str = PyUnicode_FromString(_str);
-    if (str == NULL) {
-        return NULL;
-    }
-    expr_ty the_str = _PyAST_Constant(str, NULL, b->lineno,
-                              b->col_offset, b->end_lineno,
-                              b->end_col_offset, p->arena);
-    if (the_str == NULL) {
-        Py_DECREF(str);
-        return NULL;
+    if (!is_b_empty) {
+        PyObject* str = PyUnicode_FromString(_str);
```

Comment:
I think we are leaking (and before as well) this. IIRC this needs to be added to the arena (`_PyArena_AddPyObject`) otherwise nobody is going to clean it

Reply:
Fixed in https://github.com/we-like-parsers/cpython/pull/164/commits/a257a531cff008931e9155092d3fc712761c99b8.